alua.c

```diff
@@ -776,6 +776,25 @@ int alua_check_state(struct tcmu_device *dev, struct tcmulib_cmd *cmd)
 {
 	struct tcmur_device *rdev = tcmu_dev_get_private(dev);
 
+	if (rdev->failover_type == TCMUR_DEV_FAILOVER_NOT_SET) {
+		struct list_head group_list;
+
+		/*
+		 * Normally the initiator will send an INQUIRY when a
+		 * path is discovered. If tcmu-runner restarts before the
+		 * multipath layer removes a path, then it might start
+		 * sending IO first. Run tcmu_get_alua_grps here if it's not
+		 * been run yet so the failover_type gets set and the lock
+		 * state gets updated.
+		 */
+		list_head_init(&group_list);
+		if (tcmu_get_alua_grps(dev, &group_list))
+			return TCMU_STS_HW_ERR;
+		tcmu_release_alua_grps(&group_list);
+
+		tcmu_update_dev_lock_state(dev);
+	}
+
```

Comment:
Will this happen ?

In `dev_added()` it will always do
```
1028         list_head_init(&group_list);        
1029         tcmu_get_alua_grps(dev, &group_list);
1030         tcmu_release_alua_grps(&group_list);
```

ahead of 

```
1040         ret = pthread_create(&rdev->cmdproc_thread, NULL, tcmur_cmdproc_thread,
1041                              dev);
1042         if (ret) {            
1043                 ret = -ret;   
1044                 goto cleanup_lock_cond;             
1045         }
```

Before the `tcmur_cmdproc_thread()` thread run, no any IO command could be received.